src/Analyzer.Utilities/Extensions/StringExtensions.cs

```diff
@@ -0,0 +1,18 @@
+﻿using System;
+
+namespace Analyzer.Utilities.Extensions
+{
+    public static class StringExtensions
+    {
+        public static bool IsPlural(this string word)
+        {
+            if (!word.EndsWith("s", StringComparison.Ordinal) &&
```

Comment:
Good catch, it should be OrdinalIgnoreCase , have fixed that and added testcases to cover the same. 

The plural check is definitely not enough and that's the reason I was looking at pluralizationservice (https://msdn.microsoft.com/en-us/library/system.data.entity.design.pluralizationservices.pluralizationservice(v=vs.110).aspx ) but since it's a portal lib , can't use the same. So currently doing exactly how Fxcop implemented. ( Thanks sri for pointing the link http://index/#NamingRules/FlagsEnumsShouldHavePluralNames.cs ) . We can invest on getting the pluralization perfect but that would be writing analyzer for pluralization.

Regarding if words ending with 'i' and 'ae' , I believe they belong to irregular plurals(https://en.wiktionary.org/wiki/Category:English_irregular_plurals_ending_in_%22-i%22)